### PR TITLE
[12.0] schedule improvements

### PR DESCRIPTION
**Pull request by NL66278**
Optionally adjust next_run when it appears scheduled runs have been missed.

### CODE REVIEW

File: account_bank_statement_import_online/models/online_bank_statement_provider.py [hunk 1]
```diff
@@ -107,6 +108,10 @@ class OnlineBankStatementProvider(models.Model):
     certificate_private_key = fields.Text()
     certificate_chain = fields.Text()
     allow_empty_statements = fields.Boolean()
+    adjust_next_run = fields.Boolean(
```
Comment (alexey-pelykh) [inline account_bank_statement_import_online/models/online_bank_statement_provider.py]: It's safe to have this always enabled, don't think there's need for an option tbh

File: account_bank_statement_import_online/tests/test_account_bank_statement_import_online.py [hunk 2]
```diff
@@ -267,6 +267,48 @@ def test_interval_type_hours(self):
...
+        provider.interval_type = 'hours'
+        provider.adjust_next_run = True
+        five_minutes_ago = self.now - relativedelta(minutes=5)
+        three_hours_earlier = five_minutes_ago - relativedelta(hours=3)
+        fiftyfive_minutes_later = five_minutes_ago + relativedelta(hours=1)
+        provider.next_run = three_hours_earlier
+        provider.last_successful_run = three_hours_earlier - relativedelta(hours=1)
+        provider._scheduled_pull()
+        self.assertEqual(provider.last_successful_run, five_minutes_ago)
+        self.assertEqual(provider.next_run, fiftyfive_minutes_later)
```
Comment (alexey-pelykh) [inline account_bank_statement_import_online/tests/test_account_bank_statement_import_online.py:288]: To be on the super-safe size, please also check that number of pulled statement lines matches the expectation in both cases, whatever that number is

File: account_bank_statement_import_online/models/online_bank_statement_provider.py (hunk 1 above)
Comment (NL66278) [reply account_bank_statement_import_online/models/online_bank_statement_provider.py]: I will leave out the boolean

File: account_bank_statement_import_online/tests/test_account_bank_statement_import_online.py (hunk 2 above)
Comment (NL66278) [reply account_bank_statement_import_online/tests/test_account_bank_statement_import_online.py:288]: Little bit complicated as I check for hours of delay but statements are created daily, and the number of statements created might depend on the time the tests run (so either 1 or 2 actual dates might be involved).

**Comment by NL66278** [comment]
@alexey-pelykh Can you live with the PR as it is now, or do you think the test for number of statements is absolutely needed?

**Comment by NL66278** [comment]
@alexis-via Could you also have a look at this one please?

**Comment by alexey-pelykh** [comment]
@NL66278 same comment on commit message :)

**Comment by NL66278** [comment]
@alexey-pelykh Also adjusted this commit message.

**Comment by NL66278** [comment]
Closed in favor of more complete update here: https://github.com/OCA/bank-statement-import/pull/590

**Pull request closed without merging**